mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,330 @@
+        self._content = pd.read_csv(local_artifact_path)
+        return self._content
+
+
+_DEFAULT_SAMPLE_ROWS_FOR_SHAP = 2000
+
+
+class DefaultEvaluator(ModelEvaluator):
+    def can_evaluate(self, model_type, evaluator_config=None, **kwargs):
+        return model_type in ["classifier", "regressor"]
+
+    def _log_metrics(self, run_id, metrics, dataset_name):
+        """
+        Helper method to log metrics into specified run.
+        """
+        client = mlflow.tracking.MlflowClient()
+        timestamp = int(time.time() * 1000)
+        client.log_batch(
+            run_id,
+            metrics=[
+                Metric(key=f"{key}_on_{dataset_name}", value=value, timestamp=timestamp, step=0)
+                for key, value in metrics.items()
+            ],
+        )
+
+    def _log_image_artifact(
+        self, artifacts, temp_dir, do_plot, run_id, artifact_name, dataset_name
+    ):
+        client = mlflow.tracking.MlflowClient()
+        pyplot.clf()
+        do_plot()
+        artifact_file_name = f"{artifact_name}_on_{dataset_name}.png"
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pyplot.savefig(artifact_file_local_path)
+        client.log_artifact(run_id, artifact_file_local_path)
+        artifact = ImageEvaluationArtifact(uri=get_artifact_uri(run_id, artifact_file_name))
+        artifact.load(artifact_file_local_path)
+        artifacts[artifact_file_name] = artifact
+
+    def _log_pandas_df_artifact(
+        self, artifacts, temp_dir, pandas_df, run_id, artifact_name, dataset_name
+    ):
+        client = mlflow.tracking.MlflowClient()
+        artifact_file_name = f"{artifact_name}_on_{dataset_name}.csv"
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pandas_df.to_csv(artifact_file_local_path, index=False)
+        client.log_artifact(run_id, artifact_file_local_path)
+        artifact = CsvEvaluationArtifact(
+            uri=get_artifact_uri(run_id, artifact_file_name),
+            content=pandas_df,
+        )
+        artifact.load(artifact_file_local_path)
+        artifacts[artifact_file_name] = artifact
+
+    def _log_model_explainality(
+            self, artifacts, temp_dir, model, X, dataset_name, feature_names, run_id, evaluator_config
+    ):
+        sample_rows = evaluator_config.get('explainality.nsamples', _DEFAULT_SAMPLE_ROWS_FOR_SHAP)
+        algorithm = evaluator_config.get('explainality.algorithm', None)
+
+        model_loader_module = model.metadata.flavors['python_function']["loader_module"]
+
+        predict_fn = model.predict
+
+        if model_loader_module == 'mlflow.sklearn':
+            raw_model = model._model_impl
+        elif model_loader_module == 'mlflow.lightgbm':
+            raw_model = model._model_impl.lgb_model
+        elif model_loader_module == 'mlflow.xgboost':
+            raw_model = model._model_impl.xgb_model
+        else:
+            raw_model = None
```

Comment:
Can we try / catch this in case raw model resolution fails unexpectedly? Can we add a test case for this too?